Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift

```diff
@@ -146,6 +135,13 @@ class SentryFramesTrackerTests: XCTestCase {
     }
 
     func testFrameRateChange() throws {
+#if os(iOS) || os(macOS) || targetEnvironment(macCatalyst)
+        let hub = TestHub(client: nil, andScope: nil)
+        let tracer = SentryTracer(transactionContext: TransactionContext(name: "test transaction", operation: "test operation"), hub: hub)
+        
+        // the profiler must be running for the frames tracker to record frame rate info etc, validated in assertProfilingData()
+        SentryLegacyProfiler.start(withTracer: tracer.traceId)
```

Comment:
Starting the profiler is only required for this test. As running the profiler requires a hub and a trace it interfered with the test cases I added for this PR I did this change in this PR. Furthermore, it makes sense to only run the profiler for tests that need it.

Reply:
I'm good with this, thanks for pointing it out!